### PR TITLE
Dropdown

### PR DESCRIPTION
**Pull request by bjrne**
Show a dropdown or modal segue for horizontal in-course navigation

### CODE REVIEW

File: xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    @IBOutlet weak var containerView: UIView!
+    @IBOutlet weak var titleView: UILabel!
+
+    var containerContentViewController: UIViewController?
+    var course: Course!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        updateContainerView(0)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(switchViewController), name: NotificationKeys.dropdownCourseContentKey, object: nil)
```
Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift]: Matching `removeObserver` call is missing.

File: xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            vc.view.removeFromSuperview()
+            vc.removeFromParentViewController()
+            containerContentViewController = nil
+        }
+
+        let storyboard = UIStoryboard(name: "TabCourses", bundle: nil)
+        switch position {
+        case 0:
+            let vc = storyboard.instantiateViewControllerWithIdentifier("CourseContentTableViewController") as! CourseContentTableViewController
+            vc.course = course
```
Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift]: Please put the next lines in a helper function

File: xikolo-ios/ViewControllers/Courses/DropdownTableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+//  DropdownTableViewController.swift
+//  xikolo-ios
+//
+//  Created by Bjarne Sievers on 13.09.16.
+//  Copyright © 2016 HPI. All rights reserved.
+//
+
+import UIKit
+
+class DropdownTableViewController: UITableViewController {
```
Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/DropdownTableViewController.swift]: What's this VC for?

Comment (bjrne) [inline xikolo-ios/ViewControllers/Courses/DropdownTableViewController.swift]: When a user is on the course item table view page, he can now click on the navigation bar and gets a dropdown containing a tableview, where he can choose between "Learnings", "Discussion" and "Course Details". This is the VC for this static tableview.

File: xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        if let vc = containerContentViewController {
+            vc.willMoveToParentViewController(nil)
+            vc.view.removeFromSuperview()
+            vc.removeFromParentViewController()
+            containerContentViewController = nil
+        }
+
+        let storyboard = UIStoryboard(name: "TabCourses", bundle: nil)
+        switch position {
+        case 0:
```
Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift]: Please create an enum instead of using magic numbers for the position.

Comment (bjrne) [inline xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift]: @rhlff already refactored this in his dashboard branch. See e6cfaa8.

Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/CourseDecisionViewController.swift]: Cool, please copy those changes here.

**Pull request merged**